### DISCUSSION TITLE
Will the pre-1.28 JetStream API still be maintained?

### DISCUSSION THREAD

**Discussion by guojianyu** [nats-io/nats.go#1356] [General]
The new Jetstream API is officially introduced in 1.28,API rules are much clearer and more user friendly.It's a very good improvement.
but we've done a lot of work with the old APIS,do you have any good suggestions for maintenance?

**Comment by jnmoyne** [comment]
What do you mean by maintenance? For new applications, I would use the new API, for existing ones the migration is not too bad (in my initial experience) but it's best to make the move and refactoring of an application all at once because if you have the same code that needs to use both the old and the new API at the same time it can be a bit annoying and you have to duplicate some things. The old and new APIs are 'compatible' because the new API just creates pull consumers underneath the covers so 'old' and 'new' versions of the application can interoperate (as long as you're using pull consumers on the 'old' version).

**Reply by guojianyu** [reply]
Thank you for response.I will experience the new interface and try to migrate.

**Comment by wallyqs** [comment]
yes during the same v1 version of the nats.go package (following the semver rules), also the NATS Server tests themselves still use the previous APIs a lot for example. Internally both APIs are using the same JS protocol as well though the new JetStream API focuses on pull consumers only too, so when adopting one first step is to try the new style for pull consumers first.

**Reply by guojianyu** [reply]
Thank you for response.this is a good improvement,expect the new interface to provide more surprising features.
